Description: Add missing glib headers and platform defines to the tests
 This also includes adding macros and constants that present in the
 GIR data but not present in C headers on certain platforms.
Author: Zixing Liu <zixing.liu@canonical.com>
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/rust-glib-sys/+bug/2061202
Forwarded: https://github.com/gtk-rs/gtk-rs-core/pull/1361
Last-Update: 2024-04-16
---
Index: rust-glib-sys/tests/manual.h
===================================================================
--- rust-glib-sys.orig/tests/manual.h
+++ rust-glib-sys/tests/manual.h
@@ -3,6 +3,23 @@
 #include <glib.h>
 #include <glib-object.h>
 
+#ifndef G_PLATFORM_WIN32
+// glib-unix.h is not included automatically
+#include <glib-unix.h>
+// polyfill the WIN32 constants on non-Win32 platforms
+typedef enum
+{
+  G_WIN32_OS_ANY,
+  G_WIN32_OS_WORKSTATION,
+  G_WIN32_OS_SERVER,
+} GWin32OSType;
+#endif
+
+// polyfill when the platform doesn't define tracing macros
+#ifndef G_TRACE_CURRENT_TIME
+#define G_TRACE_CURRENT_TIME 0
+#endif
+
 // included in Gir on all platforms even though it is only present on windows
 #ifndef G_WIN32_MSG_HANDLE
 #define G_WIN32_MSG_HANDLE 19981206
